File: src/web.admin/Deliscio.Web.Admin/wwwroot/tsx/apis/api-response.ts
/**
 * Interface representing a basic API response object with isSuccess and message
 */
export interface ApiResponse {
    isSuccess: boolean;
    message: string | null;
}

export interface ApiDataResponse<T> extends ApiResponse {
    data: T | null;
}

/*
    export interface ApiDataResponse<T> {
    isSuccess: boolean;
    message: string | null;
    data: T | null;
    }
*/

File: src/web.admin/Deliscio.Web.Admin/Models/Responses/ApiResponse.cs
namespace Deliscio.Web.Admin.Models.Responses;

public record ApiResponse
{
    public bool IsSuccess { get; set; }

    public string Message { get; set; } = string.Empty;
}

public record ApiDataResponse<TData> : ApiResponse
{
    public TData? Data { get; set; }
}